src/miniscript/mod.rs

```diff
@@ -874,6 +874,12 @@ mod tests {
              OP_PUSHNUM_1\
              )"
         );
+
+        // Thresh bug with equal verify roundtrip
+        roundtrip(
+            &ms_str!("tv:thresh(1,pk(02d7924d4f7d43ea965a465ae3095ff41131e5946f3c85f79e44adbcf8e27e080e))", ),
+            "Script(OP_PUSHBYTES_33 02d7924d4f7d43ea965a465ae3095ff41131e5946f3c85f79e44adbcf8e27e080e OP_CHECKSIG OP_PUSHNUM_1 OP_EQUALVERIFY OP_PUSHNUM_1)",
+        );
```

Comment:
I seem to only be able to reproduce it with a single subpolicy, which is not a valid Miniscript (see https://github.com/rust-bitcoin/rust-miniscript/pull/242 ).

Also, a broader Miniscript question: why isn't the Script here `1 <pk> CHECKSIG ADD 1 EQUALVERIFY 1` (as i would have expected of a `thresh` wrapped in a `v`, wrapped in a `and_or(X, 1)`) ?

Reply:
Sorry, I have been away for a while. Will catch up with all the miniscript stuff this week.

Reply:
`and_or` is a ternary fragment(3 args) and this is binary fragment(2 args). Why should we have `and_or` here?

Reply:
Sorry, i meant `and_v`. Editing now.

Reply:
And i think i answered the question myself, `ADD` is only appended if `k > 1`.